classes/manage_content_types.php

```diff
@@ -352,6 +353,7 @@ public function update(int $id, \stdClass $data) {
         $cleandata = cms_types::clean_record($data);
         $instance->from_record($cleandata);
         $instance->update();
+        lib::reset_cms_names($id);
```

Comment:
I can see this being a performance issue on big sites. Is there any way we can invalidate this and not recalculate them until they are needed just in time?

Reply:
As far as I know, the name gets pulled directly from the database, not from an API or cache. The label  function get_label_name() is not part of an API, but gets called from add_instance() and update_instance().

We could submit a ticket. In the meantime, perhaps make it an ad-hoc task?

Reply:
Lets ship it as is and worry about it when we have a real site that has slow updates